code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/DrawableResourceGenerator.kt

```diff
@@ -44,32 +42,48 @@ class DrawableResourceGenerator (
     )
 
     override fun generateResource(
-            number: Int,
-            outputFolder: File,
-            resourceQualifiers: List<String>,
-            resourceExtension: String
+            properties: AbstractAndroidResourceProperties,
+            outputFolder: File
     ) {
-        repeat(number) {
+        // Sanity check
+        if (properties.propertyType != ResourcePropertyType.DEFAULT) {
+            throw RuntimeException ("Unexpected property type. Got ${properties.propertyType} instead of ${ResourcePropertyType.DEFAULT}")
+        }
+        (properties as DefaultAndroidResourceProperties).fileData.forEach { fileData ->
             // TODO: generate unique IDs only by qualifiers or folder name so the same resource appears on hdpi and mdpi
-            when (resourceExtension) {
+            when (properties.extension) {
                 "xml" ->  {
-                    val outputFile = File(outputFolder, "vector_drawable_${uniqueIdGenerator.genIdByCategory("drawable.fileName.vectorDrawable")}.${resourceExtension}")
+                    val fileName = "vector_drawable_${params.uniqueIdGenerator.genIdByCategory("drawable.fileName.vectorDrawable.${properties.qualifiers}")}"
+                    val outputFile = File(outputFolder, "$fileName.${properties.extension}")
                     println("Generating ${outputFile.absolutePath}")
-                    generateVectorDrawableResource(outputFile, resourceQualifiers)
+                    generateVectorDrawableResource(outputFile, properties.splitQualifiers, fileData)
+                    params.resourceModel.resourceList.add(ResourceData(
+                        pkg = "",
+                        name = fileName,
+                        type = "drawable",
+                        extension = properties.extension,
+                        qualifiers = properties.splitQualifiers))
                 }
                 else -> {
-                    val outputFile = File(outputFolder, "image_${uniqueIdGenerator.genIdByCategory("drawable.fileName.image")}.${resourceExtension}")
+                    val fileName = "image_${params.uniqueIdGenerator.genIdByCategory("drawable.fileName.image.${properties.qualifiers}")}"
+                    val outputFile = File(outputFolder, "$fileName.${properties.extension}")
                     println("Generating ${outputFile.absolutePath}")
-                    generateImageResource(outputFile, resourceQualifiers, resourceExtension)
+                    generateImageResource(outputFile, properties.extension, fileData)
+                    params.resourceModel.resourceList.add(ResourceData(
+                        pkg = "",
```

Comment:
use default value ?

Reply:
This is just a placeholder for now. It won't have a default value if I end up needing it.